conn.go

```diff
@@ -339,6 +339,15 @@ func DialOpen(d Dialer, name string) (_ driver.Conn, err error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// cn.ssl and cn.startup panic on error. Make sure we don't leak cn.c.
+	panicking := true
+	defer func() {
+		if panicking {
```

Comment:
Since `err` is a named return value I *think* you can just do `if err != nil` and drop `panicking`.

Reply:
I thought so, too, initially, but you can't. This defer runs before the recover-defer, so err will still be nil here.

Reply:
Ugh. I'm not aware of a project that is more popular and more un-Goish than lib/pq.